### PR TITLE
enable installation of etcd-kubernetes-resources-count-exporter for CAPI

### PR DESCRIPTION
**Pull request by njuettner**
Issue: https://github.com/giantswarm/giantswarm/issues/26846

Unfortunately we need to change permissions (running as root) because the certs from mount volume are only readable for root. I tested several things but nothing was really working besides changing permissions.

Also in CAPI we have a different hostPath where the etcd certs are located, changing the hostPath is already allowed by https://github.com/giantswarm/etcd-kubernetes-resources-count-exporter/pull/63. 

I also ensured it is possible with Kyverno Policies once migrated for v1.25.

Tested in CAPI cluster with this configuration:

```yaml
 etcd:
  hostPath: "/etc/kubernetes/pki/etcd/"
  cacertpath: "/certs/ca.crt"
  certpath: "/certs/server.crt"
  keypath: "/certs/server.key"
```

### CODE REVIEW

File: helm/etcd-kubernetes-resources-count-exporter/values.yaml [hunk 1]
```diff
@@ -56,3 +56,7 @@ podSecurityContext:
 securityContext:
   seccompProfile:
     type: RuntimeDefault
+
+kyvernoPolicyExceptions:
```
Comment (whites11) [inline helm/etcd-kubernetes-resources-count-exporter/values.yaml:60]: Wouldn't this make the app not work in legacy clusters?

File: helm/etcd-kubernetes-resources-count-exporter/templates/policy-exceptions.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{{- if .Values.kyvernoPolicyExceptions.enabled }}
+  {{- if .Capabilities.APIVersions.Has "kyverno.io/v2alpha1/PolicyException" -}}
+apiVersion: kyverno.io/v2alpha1
+kind: PolicyException
+metadata:
+  name: {{ tpl .Values.resource.default.name . }}-policy-exceptions
+  namespace: {{ .Values.kyvernoPolicyExceptions.namespace | default "kube-system" }}
```
Comment (stone-z) [inline helm/etcd-kubernetes-resources-count-exporter/templates/policy-exceptions.yaml:7]:
```suggestion
  namespace: {{ .Values.kyvernoPolicyExceptions.namespace | default "giantswarm" }}
```

File: helm/etcd-kubernetes-resources-count-exporter/values.yaml (hunk 1 above)
Comment (stone-z) [reply helm/etcd-kubernetes-resources-count-exporter/values.yaml:60]: With the [capability gate](https://github.com/giantswarm/etcd-kubernetes-resources-count-exporter/pull/64/files#diff-d45ffff7f0d3917926869f5fa5db1614e826d86d5d57e819a06217aa9d0fa526R2), the polex wouldn't be created in clusters without Kyverno.

My slight concern is for cases where this is installed into a cluster without Kyverno and then Kyverno is later installed. The PolicyException would not be created until this app is deployed again, so there is some time where the pods could be rejected if policies are being enforced. This is a general problem Shield is working on so I would suggest this PR can continue and we'll keep working on a more general solution.

File: helm/etcd-kubernetes-resources-count-exporter/templates/policy-exceptions.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  labels:
+    {{- include "etcd-kubernetes-resources-count-exporter.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": "pre-install,pre-upgrade"
+spec:
+  exceptions:
+  - policyName: require-run-as-non-root-user
+    ruleNames:
+    - run-as-non-root-user
```
Comment (stone-z) [inline helm/etcd-kubernetes-resources-count-exporter/templates/policy-exceptions.yaml:16]: Suggestion: according to [the dashboard](https://giantswarm.grafana.net/d/e5SZJRo4z/security-teams-overview?orgId=1&var-team=All&var-category=All&var-policy=All&var-app=etcd-kubernetes-resources-count-exporter&var-deployment=All&var-daemonset=All), this app is failing some other policies. Some of them should just be fixed in the chart itself, but others may also need exceptions. You might consider just getting that out of the way in this PR.

**Review by stone-z — APPROVED** [review]
This is fine for me as a starting point. I left one suggestion but it can be handled outside the scope of this PR

**Review by whites11 — APPROVED** [review]

File: helm/etcd-kubernetes-resources-count-exporter/templates/policy-exceptions.yaml (hunk 3 above)
Comment (njuettner) [reply helm/etcd-kubernetes-resources-count-exporter/templates/policy-exceptions.yaml:16]: I added more exceptions, I was able to verify we should have everything now. Thanks for pointing me to the dashboard. 

I tried to limit the exceptions as low as possible.

File: helm/etcd-kubernetes-resources-count-exporter/values.yaml [hunk 4]
```diff
@@ -6,9 +6,9 @@ image:
   tag: "[[ .Version ]]"
 pod:
   user:
-    id: 1000
+    id: 0
```
Comment (erkanerol) [inline helm/etcd-kubernetes-resources-count-exporter/values.yaml:9]: When I checked this issue, I couldn't find any solution other than this. Therefore, it is OK.

**Review by erkanerol — APPROVED** [review]

**Review by stone-z — APPROVED** [review]

**Pull request merged**